### PR TITLE
feat(Input fields): update iconButton in all fields

### PR DESCRIPTION
**Pull request by marcoskolodny**
Issue: [Link](https://jira.tid.es/browse/WEB-1762)

### CODE REVIEW

File: src/__screenshot_tests__/__image_snapshots__/double-field-screenshot-test-tsx-double-field-3-snap.png
Comment (marcoskolodny) [inline src/__screenshot_tests__/__image_snapshots__/double-field-screenshot-test-tsx-double-field-3-snap.png:1]: Expected change: the button's area has increased

File: src/text-field-base.tsx [hunk 1]
```diff
@@ -29,35 +30,29 @@ const isValidInputValue = (value?: string, inputType?: React.HTMLInputTypeAttrib
 
 interface FieldEndIconProps {
     Icon: React.FC<IconProps>;
-    className?: string;
+    /** In date fields, we want the icon's background to stay transparent when hovering/pressing it */
+    hasBackgroundColor?: boolean;
```
Comment (atabel) [inline src/text-field-base.tsx:34]: isn't this a design inconsistency?

**Review by atabel — APPROVED** [review]

File: src/text-field-base.tsx (hunk 1 above)
Comment (pladaria) [reply src/text-field-base.tsx:34]: In this field the icon is decorative. The whole field is clickable to show the picker

**Review by pladaria — APPROVED** [review]

File: src/password-field.tsx [hunk 2]
```diff
@@ -21,11 +20,11 @@ const PasswordAdornment: React.FC<{
     const {texts} = useTheme();
     return (
         <FieldEndIcon
-            Icon={isVisible ? IconEyeOffRegular : IconEyeRegular}
-            aria-label={texts.togglePasswordVisibilityLabel}
-            className={endIconWithShadow}
-            onPress={() => {
-                setVisibility(!isVisible);
+            checkedProps={{Icon: IconEyeOffRegular, 'aria-label': texts.togglePasswordVisibilityLabel}}
```
Comment (aweell) [inline src/password-field.tsx:23]: Deberíamos dividir `togglePasswordVisibilityLabel` para que cuando esté oculta dicte: "Mostrar contraseña" y cuando este visible "Ocultar contraseña"?

Comment (marcoskolodny) [reply src/password-field.tsx]: Good idea. Done

**Review by aweell — APPROVED** [review]

File: src/theme.tsx [hunk 3]
```diff
@@ -147,7 +150,8 @@ const TEXTS_DE: ThemeTexts = {
 
 const TEXTS_PT: ThemeTexts = {
     expirationDatePlaceholder: 'MM/AA',
-    togglePasswordVisibilityLabel: 'Mostrar ou ocultar senha',
+    enablePasswordVisibility: 'mostrar senha',
+    disablePasswordVisibility: 'esconder a senha',
     loading: 'Carregando',
```
Comment (yceballost) [inline src/theme.tsx:155]: @nazaretcaballo-CX can you check these texts? 🫶

**Review by yceballost — APPROVED** [review]

**Pull request merged**